src/main/resources/com/parasoft/findings/jenkins/coverage/cobertura.xsl

```diff
@@ -40,8 +87,13 @@
 
     <xsl:template name="packages">
         <xsl:element name="packages">
-<!--             Group by the parent path of uri-->
+            <!-- Group by the parent path of uri-->
             <xsl:for-each-group select="/Coverage/Locations/Loc" group-by="substring-before(@uri, tokenize(@uri, '/')[last()])">
+                <xsl:variable name="uriWithoutFilePrefix">
```

Comment:
Here we are in a group, which is grouped based on the parent directory of the files. The `uri` used here corresponds to the first file in the group.

When calculating `filePath` later at `class` level, using this `uriWithoutFilePrefix` will cause the `filePath` for all files in the group to be the first file in the group, leading to incorrect results.

You can use the coverage report generated by bank example as a test case to verify this.